### ISSUE TITLE
No module factory available for dependency type: CssDependency on v3.1.3

### ISSUE THREAD

**Issue by JavRok** [jaredpalmer/razzle#1314]
### Current Behavior

When upgrading razzle from 3.0.0 to 3.1.3, I can no longer `razzle build` due to error:

```
Failed to compile.

No module factory available for dependency type: CssDependency
```
This happens on the `Compiling client...` step. And I think is related to this thread in mini-css-extract-plugin:
https://github.com/webpack-contrib/mini-css-extract-plugin/issues/493

The local webpack dev server works. It seems that it doesn't find the mentioned plugin only for build environment. 

### Expected behavior

It should compile the same way it does for version 3.0.0. 

### Reproducible example

I have a config using razzle-plugin-scss, haven't tried the official example, but I'd like to know if anybody else is experiencing this issue. 

### Suggested solution(s)

In the mentioned thread there's some possible solutions:
https://github.com/webpack-contrib/mini-css-extract-plugin/issues/493

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Razzle            |  3.1.3
| Razzle Plugins   |  3.1.3
| Node             |  v13.8.0
| Browser          |  Node
| npm/Yarn         |  npm 6.13.6
| Operating System |  WSL Ubuntu
| TypeScript       |  3.9.3
| React            |  16.8.6

**Comment by fivethreeo** [comment]
If you got docker, try:

```bash
sudo docker run -it --rm --mount src=`pwd`,target=/test_container,type=bind --user root node:13-alpine /bin/sh
```
Inside docker container run

```bash
apk add --no-cache --virtual .build-deps build-base python libsass musl-dev git curl
cd /test_container
curl https://codeload.github.com/jaredpalmer/razzle/tar.gz/master | tar -xz --strip=2 razzle-master/examples/with-scss
cd with-scss
yarn install
yarn build
yarn start
```

**Comment by fivethreeo** [comment]
Um, WSL is not needed for razzle. It supports windows :)

**Comment by fivethreeo** [comment]
If you used npm try using yarn. This seems like you load two different packages in webpack loader and plugin. Maybe scss gets its own copy. So might be WSL doing something strange. Or we need to skip the dep or make it a peerDependency so scss plugin uses the one supplied by razzle core.

**Comment by jamesvillarrubia** [comment]
Downgrading mini-css-extract-plugin to 0.8.0 inside razzle-plugin-scss will resolve the problem.  If you want, I've forked the plugin.  Just add 
```razzle-plugin-scss: "github:jamesvillarrubia/razzle-plugin-scss"```
to your package.json.

I can make a PR if it's useful to hardcode the plugin dependency at 0.8.0.

**Comment by fivethreeo** [comment]
@jamesvillarrubia what is the os and node version?

**Comment by fivethreeo** [comment]
I will try removing mini-css-extract-plugin dependency from scss plugin so we ensure there is only one instance. Was a issue for this on extract plugin.

**Comment by fivethreeo** [comment]
Tagged a new patch release now, does 3.1.4 fix this issue?

**Comment by JavRok** [comment]
Hi, sorry for the delayed answer, this is a production app and I couldn't test until now. And then it took me a while to test this properly... 

I can confirm that yarn works after the update. I upgraded razzle and plugins to 3.1.4. One of my issues is that my host node version was 13, but the docker container was 10 ... my bad. I only could upgrade it to 12 and I can confirm that razzle build works on it, **BUT only with Yarn**. 

I thought that there shouldn't be any difference of how dependencies are managed in both yarn & npm, but it seems I'm wrong.
If I run the process with npm from scratch, i.e. ``npm install && razzle build``, it will complain for the missing 'mini-css-extract-plugin' that you removed. If I install it manually, it will then go back to the same 'CssDependency' error. 

But with just ``yarn install && razzle build`` it just works.

Personally I think that a project like this should work on NPM out of the box. I think there's some kind of issue with the dependency tree. But for now this works for me. 

Cheers !

**Comment by fivethreeo** [comment]
Maybe if I add it to scss-plugin peerDependencies, would that work?

**Comment by JavRok** [comment]
It could work, but I don't have enough experience with peerDependencies, so I'm not sure if this will fix the CssDependency error. 

If you give it a try, I'll test as soon as I can

Thanks again

**Comment by fivethreeo** [comment]
New version out now :)

**Comment by fivethreeo** [comment]
This version uses peerDependencies so if npm warns about wrong dependency versions add those dependencies to resolutions in your projects package.jdon

**Comment by aforty** [comment]
Not sure if people are still having an issue but I fixed my `less` plugin (which basically works the same way as scss) by replacing the existing instance of the `MiniCssExtractPlugin` with one created from the plugin, thus there's no package conflict (webpack-contrib/mini-css-extract-plugin#493) and everything compiles as it should. It's _**super hacky**_ but seems like it needs something like this until either the plugin is fixed or everything uses it as a `peerDependency`.

You'd add this around `razzle-plugin-scss/index.js:97`

```
const miniCssPlugin = config.plugins.find(i => i.constructor.name === 'MiniCssExtractPlugin')
if (miniCssPlugin) {
  const idx = config.plugins.indexOf(miniCssPlugin);
  config.plugins.splice(idx, 1, new MiniCssExtractPlugin(miniCssPlugin.options));
}
```

**Comment by fivethreeo** [comment]
@aforty did you try adding mini-css-extract-plugin@0.9 as a resolution? 3.1.5 should fix this, also make sure the deps are not still present in node_modules

**Comment by JavRok** [comment]
After latest update, I can run the build with npm both locally and in my CI (Jenkins), so after a lot of struggle... I consider this fixed. Thanks for the help!

Note aside, in one of the commits some config was removed from css-loader (accidentally?) which I had to overwrite. I commented it here:
https://github.com/jaredpalmer/razzle/commit/fd857a242519135ce422609b3a1611705b70d113#diff-4fc6a7be00756e3dbc6b0eac82ea10e5

Cheers !
